spec/controllers/api/v1/project_preferences_controller_spec.rb

```diff
@@ -234,13 +235,17 @@
     end
 
     describe 'user specific preferences' do
-      before do
+      it 'only fetches settings of owned project' do
         default_request user_id: unauthorised_user.id, scopes: scopes
-        run_unauthorised_user_read
+        get :read_settings, params: { project_id: project.id, user_id: unauthorised_user.id, format: :json }
+        expect(response.status).to eq(404)
       end
 
-      it 'only fetches settings of owned project' do
-        expect(response.status).to eq(404)
+      it 'only fetches settings of the authorized user' do
```

Comment:
For clarity, this isn't 'authorized' user as much as it is 'specified' user, as it's the user_id specified in the params. This is unrelated to that user's authorization since users can't obtain their own UPPs anyway.